libsolidity/interface/CompilerStack.cpp

```diff
@@ -281,53 +281,55 @@ bool CompilerStack::analyze()
 	{
```

Comment:
If `source->ast == nullptr` it seems like this entire block will do nothing.
Wouldn't it be better to just stop earlier if so?

Reply:
Mmh, wouldn't that solution need to check _all_ ASTs beforehand? 
```
for (Source const* source: m_sourceOrder)
    if (!source->ast)
        return false;
```
Maybe for some source files it's fine and we could still run analysis on them?

Reply:
Ah right, I missed the `for` loops ;)